.github/workflows/compile-requirements.yml

```diff
@@ -0,0 +1,137 @@
+name: Reusable WF - Compile Pip Requirements
+
+on:
+  workflow_call:
+    inputs:
+      optional-dependencies:
+        description: An optional dependency installation specification for use with a pyproject.toml file.
+        type: string
+        required: false
+        default: ".[dev]"
+      upstream-packages:
+        description: A whitespace separated list of Howso package dependencies (e.g., "amalgam-lang-py howso-engine-py").
+        type: string
+        required: false
+
+defaults:
+  run:
+    shell: bash
+
+jobs:
+  check-requirements:
+    runs-on: ubuntu-latest
+    outputs:
+      dev_req_file: ${{ steps.detect-changes.outputs.dev-req-file }}
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          ref: ${{ github.head_ref }}
+
+      - name: Check for requirements changes
+        id: detect-changes
+        run: |
+          # Determine which type of requirements/setup file exists in this repo
+          if [ -e "pyproject.toml" ]; then
```

Comment:
Think it is worth putting a warning for repos that use setup.py letting them know to update to pyproject.toml? Might be a good place to add it.